SmartcarAuth/SmartcarAuth.swift

```diff
@@ -85,19 +85,20 @@ Smartcar Authentication SDK for iOS written in Swift 5.
     */
     public func launchAuthFlow(url: String, viewController: UIViewController? = nil) {
         let authUrl = URL(string: url)
-        let redirectUriScheme = "sc" + self.clientId
-
+        let redirectUrl = URL(string: redirectUri)
+        let redirectUriScheme = redirectUrl?.scheme
+        
         if #available(iOS 11.0, *) {
             if #available(iOS 13.0, *) {
-                let authSession = ASWebAuthenticationSession.init(url: authUrl!, callbackURLScheme: redirectUriScheme, completionHandler: webAuthSessionCompletion)
+                let authSession = ASWebAuthenticationSession.init(url: authUrl!, callbackURLScheme: redirectUriScheme!, completionHandler: webAuthSessionCompletion)
                 // this is required for the browser to open in iOS 13
                 authSession.presentationContextProvider = self
                 self.session = authSession
             } else if #available(iOS 12.0, *) {
-                let authSession = ASWebAuthenticationSession.init(url: authUrl!, callbackURLScheme: redirectUriScheme, completionHandler: webAuthSessionCompletion)
+                let authSession = ASWebAuthenticationSession.init(url: authUrl!, callbackURLScheme: redirectUriScheme!, completionHandler: webAuthSessionCompletion)
                 self.session = authSession
             } else {
-                let authSession = SFAuthenticationSession.init(url: authUrl!, callbackURLScheme: redirectUriScheme, completionHandler: webAuthSessionCompletion)
+                let authSession = SFAuthenticationSession.init(url: authUrl!, callbackURLScheme: redirectUriScheme!, completionHandler: webAuthSessionCompletion)
```

Comment:
why the `!` now and not needed before? Are just making sure that it is there now?

Reply:
Previously we were explicitly defining what the scheme was (`sc<clientId>`) so the type of `callbackURLScheme` was automatically `String`, but since now we are deriving it from the redirectUri itself, we need to unconditionally unwrap the value of `callbackURLScheme`  so that the type is `String`.